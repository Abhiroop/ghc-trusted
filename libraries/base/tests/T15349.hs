import Control.Monad.ST.Strict
import Control.Monad.Fix
import Data.STRef

foo :: ST s Int
foo = do
  ref <- newSTRef True
  mfix $ \res -> do
    x <- readSTRef ref
    if x
      then do
        writeSTRef ref False
        return $! (res + 5)
      else return 10

main :: IO ()
main = print $ runST foo
