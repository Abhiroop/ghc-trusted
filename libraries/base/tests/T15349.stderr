T15349: <<loop>>
